components/datetime/src/any/datetime.rs

```diff
@@ -166,7 +171,11 @@ impl DateTimeFormatter {
             calendar::load_lengths_for_any_calendar_kind(&crate::provider::Baked, locale, kind)?,
             locale,
             &options,
-        )?;
+        )
+        .map_err(|e| match e {
+            PatternForLengthError::Data(e) => DateTimeError::Data(e),
+            PatternForLengthError::Pattern(e) => DateTimeError::Pattern(e),
+        })?;
```

Comment:
Suggestion: you have a lot of this remapping code; it would be a lot less verbose, and I think not harmful, to just implement a `From` impl and use `?` operators.

Reply:
I'm explicitly not doing that, because I will introduce more error types in the next PR and then it will become messy.

Reply:
Also when doing this I had to recurse a lot to figure out which variants are actually feasible, I don't want the next person to have to do that as well.

Reply:
It's hard for me to envision how `?` operators make things more messy

Reply:
It's unclear at the call site what happens, you don't see what the error type from the nested call is, and which error variants it has.

Reply:
We can clean this up when we make the public-facing error changes. This code will be changed more in the short term so I'd like to keep things explicit for now.

Reply:
The fact that the internal functions have narrow error types I think is more than enough explicitness. This `try_write_field` function is extremely complex, perhaps the most complex function in the whole library, and I think the readability of using `?` is more important here. I don't care at each and every call site what the different error variants can be, as long as they map cleanly into `DateTimeWriteError`, which the `?` operator guarantees.